# ATAC_EPO_vs_Placebo

<h2>Single-cell ATAC-seq Library Processing and Alignment</h2>
Single-cell ATAC-seq libraries were processed using Cell Ranger ATAC v2.0.0 (10x Genomics). Raw sequencing reads from eight hippocampal samples (A1, A2, A4, A6, B7, B8, B9, B11) were aligned to the mm10 reference genome (refdata-cellranger-arc-mm10-2020-A-2.0.0). Fragment files and peak-barcode matrices were generated using the cellranger-atac count (2.1.0). The resulting fragments files and peak count matrices were used as input for downstream analysis in R using Signac and Seurat, as described below. 
<h3>Analysis Using a Unified Peak Set </h3>
All downstream analysis was conducted in R using the Signac (v1.13.0) and Seurat (v5.1.0) packages.  The fragments files and peak-barcode matrices output by Cell Ranger ATAC were used as input. First, peaks were loaded for each sample and a unified peak set was created by merging the peaks across all eight samples using the reduce() function from GenomicRanges. Peaks shorter than 20 bp or longer than 10,000 bp were excluded. For each sample, cells with fewer than 500 fragments in peaks were excluded and peak counts were quantified using FeatureMatrix() against the unified peak set, then, chromatin assays were created for every individual sample. 
Annotation was added from EnsDb.Mmusculus.v79. For each Seurat object, quality control (QC) metrics were computed using Signac functions to ensure high-quality nuclei were retained for downstream analysis. The following metrics were calculated: TSS enrichment, Nucleosome signal, Blacklist ratio using blacklist_mm10 from Signac, Percent reads in peaks: calculated manually as (peak_region_fragments / passed_filters) * 100. Filtering thresholds were determined per sample using quantile-based cutoffs. Cells with peak_region_fragments below the 2nd percentile or above the 98th percentile were excluded to remove low-quality cells and potential multiplets, respectively. Additional criteria included removing cells with pct_reads_in_peaks below the 2nd percentile, blacklist_ratio above the 98th percentile nucleosome_signal above the 98th percentile, and TSS.enrichment below the 2nd percentile. 
After QC, each Seurat object was annotated with a sample label and merged using merge() with add.cell.ids to ensure unique cell barcodes. The combined object was processed as follows: Normalization: RunTFIDF(), Dimensionality reduction: RunSVD() (LSI), Batch correction: RunHarmony() on LSI components (dims 2–30), Visualization: RunUMAP() on Harmony-corrected dimensions. 
Clusters were identified using FindNeighbors() and FindClusters() with resolution 0.8. Gene activity scores were computed using GeneActivity() and added as a new assay to the Seurat object. These were log-normalized using NormalizeData() with scale.factor median of nCount_ACTIVITY. 
To annotate cells, preprocessed single-nucleus RNA-seq data from Singh et al. (2023) was used as a reference. To match the same animals, A3, A5, B10, and B12 samples were excluded from the reference. Then, anchors between ATAC and RNA-seq were computed using FindTransferAnchors() with the CCA reduction, followed by TransferData() for label prediction. 
<h3>Analysis Using Sample-Specific Peak Sets </h3>
In addition to the unified peak set approach, we performed a complementary analysis using sample-specific peak sets generated by CellRanger ATAC. Peak-barcode matrices and fragment files were loaded individually for each sample. Chromatin assays were constructed with min.cells = 10 parameter, and Seurat objects were created per sample. Seurat objects were merged. 

Gene annotations were added from EnsDb.Mmusculus.v79 and QC metrics including TSS enrichment, nucleosome signal, percent reads in peaks, and blacklist ratio were computed. Cells were filtered based on the following criteria: nCount_peaks between 3,000–30,000, pct_reads_in_peaks > 15%, blacklist_ratio < 0.05, nucleosome_signal < 4, and TSS.enrichment > 3.
Dimensionality reduction and clustering are performed: Normalization via RunTFIDF(), Feature selection with FindTopFeatures(min.cutoff = 'q0'), LSI via RunSVD() (dims 2–8), UMAP projection with RunUMAP(), Clustering with FindNeighbors() and FindClusters() (algorithm = 1). Gene activity scores were calculated using GeneActivity() and log-normalized with NormalizeData(). Cell-type annotations predicted from the previous reduced-peak integration (see above) were transferred using shared cell barcodes. Cells without annotation were excluded. 

<h3>Motif Enrichment Analysis between Conditions</h3>h3>
Motif enrichment was performed using Signac with position frequency matrices from the JASPAR2020 “CORE” collection database. Motifs were added with the BSgenome.Mmusculus.UCSC.mm10 genome. Cells were grouped by cluster and treatment condition, and differential accessibility between EPO and Placebo groups was assessed per cluster using a likelihood ratio test with FindMarkers() with peaks detected in at least 10% of cells in either group (min.pct = 0.1) and a likelihood ratio test (test.use = "LR") while regressing out total accessibility (latent.vars = "nCount_peaks"). The closest genes to each cluster's DA peaks are calculated
Significantly enriched peaks (adjusted p < 0.05) were used for motif analysis. For clusters with ≥10 significant peaks, motif enrichment was performed using FindMotifs() to identify transcription factor motifs enriched in EPO-specific accessible regions.  



